substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/PointsToAnalysis.java

```diff
@@ -703,6 +704,61 @@ public HostVM getHostVM() {
         return hostVM;
     }
 
+    /**
+     * Iterate until analysis reaches a fixpoint.
+     *
+     *
+     * @param debug debug context
+     * @param duringAnalysisAction  The action handler taken during analysis, return true if no more iteration is required.
+     *                              It could be the {@link org.graalvm.nativeimage.hosted.Feature#duringAnalysis}
+     *                              for Native Image generation, but also can be other similar actions for standalone pointsto
+     *                              analysis that does not dependent on {@link com.oracle.svm.hosted.FeatureImpl.DuringAnalysisAccessImpl}
+     *                              and {@link com.oracle.svm.hosted.NativeImageGenerator.AnalysisFeatureHandler}.
+     * @return The error message during the analysis. If the analysis ends successfully, an empty String is returned, otherwise
+     * the actual error message is returned.
+     * @throws InterruptedException
+     */
+    @SuppressWarnings("try")
+    public String iterativelyAnalyze(DebugContext debug, Function<AnalysisUniverse, Boolean> duringAnalysisAction) throws InterruptedException {
+        int numIterations = 0;
+        while (true) {
+            try (Indent indent2 = debug.logAndIndent("new analysis iteration")) {
+                /*
+                 * Do the analysis (which itself is done in a similar iterative process)
+                 */
+                boolean analysisChanged = finish();
+
+                numIterations++;
+                if (numIterations > 1000) {
+                    /*
+                     * Usually there are < 10 iterations. If we have so many iterations, we
+                     * probably have an endless loop (but at least we have a performance
+                     * problem because we re-start the analysis so often).
+                     */
+                    return String.format("Static analysis did not reach a fix point after %d iterations because a Feature keeps requesting new analysis iterations. " +
+                                    "The analysis itself %s find a change in type states in the last iteration.",
+                            numIterations, analysisChanged ? "DID" : "DID NOT");
+
+                }
+
+                /*
+                 * Allow features to change the universe.
+                 */
+                try (StopTimer t2 = getProcessFeaturesTimer().start()) {
+                    int numTypes = universe.getTypes().size();
+                    int numMethods = universe.getMethods().size();
+                    int numFields = universe.getFields().size();
+                    if (duringAnalysisAction.apply(universe)) {
```

Comment:
`duringAnalysisAction` can be named `requireAnalysisIteration` to better express the intent.